From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jaden <madcapsnow25@gmail.com>
Date: Thu, 12 Jun 2025 15:53:04 -0500
Subject: [PATCH] Use more accurate sprint state


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index cc4b2061ae9cbd5a30b297be477cde703203b3f5..0b13465318a39e75d4d8a3b3539823750debcfe1 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -1520,7 +1520,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
 
     @Override
     public boolean isSprinting() {
-        return this.getHandle().isSprinting();
+        return this.getHandle().isAttackKnockbackIncreased();
     }
 
     @Override
