--- a/aspaper-server/build.gradle.kts
+++ b/aspaper-server/build.gradle.kts
@@ -142,9 +_,15 @@
     }
 }
 
+repositories {
+    maven {
+        url = uri("https://repo.infernalsuite.com/repository/maven-snapshots/")
+    }
+}
+
 dependencies {
-    implementation(project(":aspaper-api")) //ASP
-    implementation(project(":core")) //ASP
+    implementation("com.infernalsuite.asp:api:4.0.0-SNAPSHOT") //ASP
+    implementation("com.infernalsuite.asp:core:4.0.0-SNAPSHOT") //ASP
     implementation("commons-io:commons-io:2.11.0")
     implementation("ca.spottedleaf:concurrentutil:0.0.3")
     implementation("org.jline:jline-terminal-ffm:3.27.1") // use ffm on java 22+
