From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jaden <madcapsnow25@gmail.com>
Date: Thu, 29 May 2025 02:17:01 -0500
Subject: [PATCH] Use more accurate sprint state


diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index 1ba342a1a60951f828034d3ed535b577b3990bf6..87cd1e91097983912aea1b9bfd8d24e7f31ff995 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2821,6 +2821,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
     @Override
     public void setSprinting(boolean sprinting) {
         super.setSprinting(sprinting);
+        this.setAttackKnockbackIncreased(sprinting); // Hydrogen - fix sprinting state on attack
         AttributeInstance attribute = this.getAttribute(Attributes.MOVEMENT_SPEED);
         attribute.removeModifier(SPEED_MODIFIER_SPRINTING.id());
         if (sprinting) {
@@ -2828,6 +2829,18 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
         }
     }
 
+    // SparklySlimePaper start - fix sprinting state on attack
+    private boolean attackKnockbackIncreased;
+
+    public boolean isAttackKnockbackIncreased() {
+        return this.attackKnockbackIncreased;
+    }
+
+    public void setAttackKnockbackIncreased(boolean flag) {
+        this.attackKnockbackIncreased = flag;
+    }
+    // SparklySlimePaper end
+
     public float getSoundVolume() {
         return 1.0F;
     }
diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index 20687b632f2cc3f25c8f04e34cc1f6f1d0abf741..b793c69f2fbbf5afc78e9cedd8bbc39dd1c49324 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -1205,12 +1205,11 @@ public abstract class Player extends LivingEntity {
                     // CraftBukkit end
                     if (f > 0.0F || f1 > 0.0F) {
                         boolean flag = attackStrengthScale > 0.9F;
-                        boolean flag1;
-                        if (this.isSprinting() && flag) {
-                            this.sendSoundEffect(this, this.getX(), this.getY(), this.getZ(), SoundEvents.PLAYER_ATTACK_KNOCKBACK, this.getSoundSource(), 1.0F, 1.0F); // Paper - send while respecting visibility
-                            flag1 = true;
-                        } else {
-                            flag1 = false;
+                        boolean flag1 = this.level().paperConfig().misc.disableSprintInterruptionOnAttack
+                                ? this.isAttackKnockbackIncreased() && flag
+                                : this.isSprinting() && flag;
+                        if (flag1) {
+                            this.sendSoundEffect(this, this.getX(), this.getY(), this.getZ(), SoundEvents.PLAYER_ATTACK_KNOCKBACK, this.getSoundSource(), 1.0F, 1.0F);
                         }
 
                         f += weaponItem.getItem().getAttackDamageBonus(target, f, damageSource);
@@ -1267,6 +1266,8 @@ public abstract class Player extends LivingEntity {
                                 // Paper start - Configurable sprint interruption on attack
                                 if (!this.level().paperConfig().misc.disableSprintInterruptionOnAttack) {
                                 this.setSprinting(false);
+                                } else {
+                                    this.setAttackKnockbackIncreased(false);
                                 }
                                 // Paper end - Configurable sprint interruption on attack
                             }
