--- a/aspaper-api/build.gradle.kts
+++ b/aspaper-api/build.gradle.kts
@@ -38,8 +_,14 @@
     }
 }
 
+repositories {
+    maven {
+        url = uri("https://repo.infernalsuite.com/repository/maven-snapshots/")
+    }
+}
+
 dependencies {
-    api(project(":api")) //ASP
+    api("com.infernalsuite.asp:api:4.0.0-SNAPSHOT") //ASP
     // api dependencies are listed transitively to API consumers
     api("com.google.guava:guava:33.3.1-jre")
     api("com.google.code.gson:gson:2.11.0")
@@ -104,17 +_,21 @@
         java {
             srcDir(generatedApiPath)
             srcDir(file("../paper-api/src/main/java"))
+            srcDir(file("../aspaper-api/src/main/java"))
         }
         resources {
             srcDir(file("../paper-api/src/main/resources"))
+            srcDir(file("../aspaper-api/src/main/resources"))
         }
     }
     test {
         java {
             srcDir(file("../paper-api/src/test/java"))
+            srcDir(file("../aspaper-api/src/test/java"))
         }
         resources {
             srcDir(file("../paper-api/src/test/resources"))
+            srcDir(file("../aspaper-api/src/test/resources"))
         }
     }
 }
